make lxc-master failed: CAP_SYS_ADMIN undeclared

**Required information**

uname -r
2.6.32-431.el6.x86_64

**Issue description**

make failed:
start.c: In function ‘do_start’:
start.c:956: error: ‘PR_SET_NO_NEW_PRIVS’ undeclared (first use in this function)

**Steps to reproduce**

Step one
./autogen.sh - ok

Step two
./configure - ok

Step three
make - not ok
make
...
cgroups/cgfsng.c: In function ‘cgfsng_mount’:
cgroups/cgfsng.c:1687: error: ‘CAP_SYS_ADMIN’ undeclared (first use in this function)
cgroups/cgfsng.c:1687: error: (Each undeclared identifier is reported only once
cgroups/cgfsng.c:1687: error: for each function it appears in.)
make[3]: *** [cgroups/liblxc_la-cgfsng.lo] Error 1
make[3]: Leaving directory `/tmp/lxc-master/src/lxc'
make[2]: *** [all-recursive] Error 1
make[2]: Leaving directory `/tmp/lxc-master/src'
make[1]: *** [all] Error 2
make[1]: Leaving directory `/tmp/lxc-master/src'
make: *** [all-recursive] Error 1

--- a/src/lxc/conf.c
+++ b/src/lxc/conf.c
@@ -107,42 +107,6 @@
 
 lxc_log_define(lxc_conf, lxc);
 
-#if HAVE_LIBCAP
-#ifndef CAP_SETFCAP
-#define CAP_SETFCAP 31
-#endif
-
-#ifndef CAP_MAC_OVERRIDE
-#define CAP_MAC_OVERRIDE 32
-#endif
-
-#ifndef CAP_MAC_ADMIN
-#define CAP_MAC_ADMIN 33
-#endif
-#endif
-
-#ifndef PR_CAPBSET_DROP
-#define PR_CAPBSET_DROP 24
-#endif
-
-#ifndef LO_FLAGS_AUTOCLEAR
-#define LO_FLAGS_AUTOCLEAR 4
-#endif
-
-#ifndef CAP_SETUID
-#define CAP_SETUID 7
-#endif
-
-#ifndef CAP_SETGID
-#define CAP_SETGID 6
-#endif
-
-/* needed for cgroup automount checks, regardless of whether we
- * have included linux/capability.h or not */
-#ifndef CAP_SYS_ADMIN
-#define CAP_SYS_ADMIN 21
-#endif
-
 /* Define pivot_root() if missing from the C library */
 #ifndef HAVE_PIVOT_ROOT
 static int pivot_root(const char * new_root, const char * put_old)

--- a/src/lxc/utils.h
+++ b/src/lxc/utils.h
@@ -50,6 +50,42 @@
 #define __S_ISTYPE(mode, mask) (((mode)&S_IFMT) == (mask))
 #endif
 
+#if HAVE_LIBCAP
+#ifndef CAP_SETFCAP
+#define CAP_SETFCAP 31
+#endif
+
+#ifndef CAP_MAC_OVERRIDE
+#define CAP_MAC_OVERRIDE 32
+#endif
+
+#ifndef CAP_MAC_ADMIN
+#define CAP_MAC_ADMIN 33
+#endif
+#endif
+
+#ifndef PR_CAPBSET_DROP
+#define PR_CAPBSET_DROP 24
+#endif
+
+#ifndef LO_FLAGS_AUTOCLEAR
+#define LO_FLAGS_AUTOCLEAR 4
+#endif
+
+#ifndef CAP_SETUID
+#define CAP_SETUID 7
+#endif
+
+#ifndef CAP_SETGID
+#define CAP_SETGID 6
+#endif
+
+/* needed for cgroup automount checks, regardless of whether we
+ * have included linux/capability.h or not */
+#ifndef CAP_SYS_ADMIN
+#define CAP_SYS_ADMIN 21
+#endif
+
 /* Useful macros */
 /* Maximum number for 64 bit integer is a string with 21 digits: 2^64 - 1 = 21 */
 #define LXC_NUMSTRLEN64 21

